chroma_core/services/job_scheduler/job_scheduler.py

```diff
@@ -1159,11 +1159,26 @@ def create_ostpool(self, ostpool_data):
         self.progress.advance()
         return ostpool.id, command_id
 
-    def update_ostpool(self, ostpool_id, ostpool_data):
-        log.debug("Updating ostpool {} with: {}".format(ostpool_id, ostpool_data))
+    def update_ostpool(self, ostpool_data):
+        log.debug("Updating ostpool with: {}".format(ostpool_data))
         with self._lock:
-            ostpool = OstPool.objects.get(pk=ostpool_id)
-        # @@
+            ostpool = OstPool.objects.get(pk=ostpool_data["id"])
+
+            newlist = set(ostpool_data.get("osts", []))
+
+            with transaction.atomic():
+                current = set(ostpool.osts.all())
```

Comment:
Do the list operations need to be inside the atomic transaction? If possible, I think it would be more performant to move these operations outside the `with` and leave the `command_run_jobs` inside. Would that work or am I missing something with atomic transactions?